### PR TITLE
CB-19565 Create credential property to be able to skip organizational…

### PR DESCRIPTION
**Pull request by lturcsanyi**
… rule validation

SCPs with global condition keys and whitelisted accounts are not supported in the AWS Policy Simulator (known AWS bug), and may cause validation to fail. There is a new property on the credential entity to be able to skip validation errors related to these policies.

See detailed description in the commit message.

### CODE REVIEW

**Review by bbihari — APPROVED** [review]

File: environment/src/main/resources/schema/app/20221123111223_CB-19565_add_skip_org_policy_decisions_column.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- // CB-19565 Create credential property to be able to skip organizational rule validation
+-- Migration SQL that makes the change goes here.
+ALTER TABLE credential ADD COLUMN IF NOT EXISTS skiporgpolicydecisions BOOLEAN DEFAULT FALSE;
+UPDATE credential SET skiporgpolicydecisions = FALSE WHERE skiporgpolicydecisions IS NULL;
```
Comment (bergerdenes) [inline environment/src/main/resources/schema/app/20221123111223_CB-19565_add_skip_org_policy_decisions_column.sql:4]: Do you have a rough estimate of how many records are affected?

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java [hunk 2]
```diff
@@ -128,6 +128,13 @@ public void validate(AmazonIdentityManagementClient iam, CloudS3View cloudFileSy
                         String.join(", ", roles.stream().map(Role::getArn).collect(Collectors.toCollection(TreeSet::new))),
                         String.join("\n", failedActions.stream().limit(MAX_SIZE).collect(Collectors.toSet())));
 
+                if (validationErrorMessage.contains(AbstractAwsSimulatePolicyValidator.DENIED_BY_ORGANIZATION_RULE)) {
+                    validationErrorMessage = String.join(validationErrorMessage,
```
Comment (bergerdenes) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java:132]: I assume you wanted to use `String.format` or concat.
Join will not work in this case. The original message will be lost.

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsLogRolePermissionValidator.java [hunk 3]
```diff
@@ -107,6 +107,13 @@ public void validate(AmazonIdentityManagementClient iam, InstanceProfile instanc
                     String.join(", ", instanceProfile.getArn()),
                     String.join("\n", failedActions));
 
+            if (validationErrorMessage.contains(AbstractAwsSimulatePolicyValidator.DENIED_BY_ORGANIZATION_RULE)) {
+                validationErrorMessage = String.join(validationErrorMessage,
```
Comment (bergerdenes) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsLogRolePermissionValidator.java:111]: I assume you wanted to use `String.format` or concat.
Join will not work in this case. The original message will be lost.

Comment (bergerdenes) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsLogRolePermissionValidator.java:111]: Also no test case for this. (no test class)

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java (hunk 2 above)
Comment (bergerdenes) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java:132]: No test case for this.

File: cloud-consumption/src/test/java/com/sequenceiq/consumption/converter/CredentialResponseToCloudCredentialConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.service.secret.model.SecretResponse;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.environment.api.v1.credential.model.response.CredentialResponse;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+@ExtendWith(MockitoExtension.class)
+public class CredentialResponseToCloudCredentialConverterTest {
```
Comment (bergerdenes) [inline cloud-consumption/src/test/java/com/sequenceiq/consumption/converter/CredentialResponseToCloudCredentialConverterTest.java:23]: JUnit 5 test class and methods are not needed to be `public`

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: environment/src/main/resources/schema/app/20221123111223_CB-19565_add_skip_org_policy_decisions_column.sql (hunk 1 above)
Comment (lturcsanyi) [reply environment/src/main/resources/schema/app/20221123111223_CB-19565_add_skip_org_policy_decisions_column.sql:4]: I have no estimate, I assume it is less than the number of the envs.

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java (hunk 2 above)
Comment (lturcsanyi) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java:132]: Yes, good catch, thank you.

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsLogRolePermissionValidator.java (hunk 3 above)
Comment (lturcsanyi) [reply cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsLogRolePermissionValidator.java:111]: Yes, good catch, thank you. I will create the test as well.

File: cloud-consumption/src/test/java/com/sequenceiq/consumption/converter/CredentialResponseToCloudCredentialConverterTest.java (hunk 4 above)
Comment (lturcsanyi) [reply cloud-consumption/src/test/java/com/sequenceiq/consumption/converter/CredentialResponseToCloudCredentialConverterTest.java:23]: I see, thanks.

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**